nifi-nar-bundles/nifi-questdb-bundle/nifi-questdb/src/test/java/org/apache/nifi/questdb/embedded/EmbeddedDatabaseManagerTest.java

```diff
@@ -0,0 +1,397 @@
+
+        final Client client1 = testSubject1.acquireClient();
+        client1.insert(EVENT_TABLE_NAME, QuestDbTestUtil.getEventTableDataSource(testData));
+        final List<Event> result1 = client1.query(SELECT_QUERY, RequestMapping.getResultProcessor(QuestDbTestUtil.EVENT_TABLE_REQUEST_MAPPING));
+        client1.disconnect();
+        testSubject1.close();
+
+        assertDatabaseFolderIsNotEmpty();
+
+        final DatabaseManager testSubject2 = getTestSubject();
+        final Client client2 = testSubject2.acquireClient();
+        final List<Event> result2 = client2.query(SELECT_QUERY, RequestMapping.getResultProcessor(QuestDbTestUtil.EVENT_TABLE_REQUEST_MAPPING));
+        client2.disconnect();
+        testSubject2.close();
+
+        assertIterableEquals(result1, result2);
+    }
+
+    @Test
+    public void testDatabaseRestorationAfterLostDatabase() throws DatabaseException, IOException {
+        final List<Event> testData = getTestData();
+        final DatabaseManager testSubject = getTestSubject();
+        final Client client = testSubject.acquireClient();
+        client.insert(EVENT_TABLE_NAME, QuestDbTestUtil.getEventTableDataSource(testData));
+
+        FileUtils.deleteFilesInDir(testDbPathDirectory.toFile(), (dir, name) -> true, LOGGER, true, true);
+
+        client.insert(EVENT_TABLE_NAME, QuestDbTestUtil.getEventTableDataSource(testData));
+        final List<Event> result = client.query(SELECT_QUERY, RequestMapping.getResultProcessor(QuestDbTestUtil.EVENT_TABLE_REQUEST_MAPPING));
+
+        testSubject.close();
+
+        // Ensuring that not the fallback client answers
+        assertEquals(3, result.size());
+    }
+
+    @Test
+    public void testDatabaseRestorationAfterLosingTableFiles() throws DatabaseException, IOException {
+        final List<Event> testData = getTestData();
+        final DatabaseManager testSubject = getTestSubject();
+        final Client client = testSubject.acquireClient();
+        client.insert(EVENT_TABLE_NAME, QuestDbTestUtil.getEventTableDataSource(testData));
+
+        final File eventTableDirectory = new File(testDbPathDirectory.toFile(), "event");
+        FileUtils.deleteFilesInDir(eventTableDirectory, (dir, name) -> true, LOGGER, true, true);
+        FileUtils.deleteFile(eventTableDirectory, LOGGER);
+
+        client.insert(EVENT_TABLE_NAME, QuestDbTestUtil.getEventTableDataSource(testData));
+        final List<Event> result = client.query(SELECT_QUERY, RequestMapping.getResultProcessor(QuestDbTestUtil.EVENT_TABLE_REQUEST_MAPPING));
+
+        testDbPathDirectory.toFile().list((dir, name) -> dir.isDirectory());
+
+        testSubject.close();
+
+        // Ensuring that not the fallback client answers
+        assertEquals(3, result.size());
+    }
+
+    @Test
+    @DisabledOnOs(OS.WINDOWS)
+    // This testcase cannot be reproduced under Windows using Junit
+    public void testDatabaseRestorationAfterCorruptedFiles() throws DatabaseException, IOException {
```

Comment:
For additional reference, this is the test method that leaves the `questDbBackup` directory with `backup` sub-directories after test completion. This is probably the reason for the failure on Windows, but on both macOS and Linux, the directory remains after all tests have completed. It may be necessary to implement some manual deletion logic to ensure the directory is cleared. If the directory is cleared on macOS and Linux, it might be worth enabling this test on Windows again, as it more clearly surfaces the problem.

Reply:
The latest updates in https://github.com/apache/nifi/pull/8152/commits/e33dbe91571ef3373f356f028641f366cb4bb6c7 now clear out the `questDbBackup` directory, but the directory itself still remains. It looks like adding one more step to delete the directory itself should resolve this particular issue.

Reply:
Good catch, however the folder was left behind a different test case. I double checked if any leaves anything behind in multiple platform and I found that working well. Thanks for the finding!